core/ibft.go

```diff
@@ -1099,19 +1119,33 @@ func (i *IBFT) validPC(
 		return false
 	}
 
+	//Make sure all have the same round
+	if !messages.AllHaveSameRound(allMessages) {
+		return false
+	}
+
 	// Make sure the proposal message is sent by the proposer
 	// for the round
 	proposal := certificate.ProposalMessage
 	if !i.backend.IsProposer(proposal.From, proposal.View.Height, proposal.View.Round) {
 		return false
 	}
 
+	// Make sure that the proposal sender is valid
+	if !i.backend.IsValidSender(proposal) {
+		return false
+	}
```

Comment:
Since you need to execute this check for the proposer _and_ the prepare messages, you can probably extract a helper to check them all, for example `func allHaveValidSender(messages []*proto.Message, ...)`. 

I'm assuming you'd need to do this nested check elsewhere as well (for RC nested messages in RCC).

Reply:
Would love to keep this as explicit as possible for now and wait for a slightly bigger need for it.